### PR TITLE
WooCommerce Checkout styles

### PR DESCRIPTION
**Pull request by claudiulodro**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR styles and sets up the checkout page to more closely match our designs. Here is what it looks like:

newspack test_checkout_

I'm not super happy with the font-sizes and vertical rhythm; it doesn't feel harmonious to me, but I'm not sure what improvement to make. @laurelfulford what does your discerning eye think?

### How to test the changes in this Pull Request:

**Thorough version**: Get and build latest master branch of newspack-blocks and https://github.com/Automattic/newspack-plugin/pull/225 branch of newspack-plugin (if it hasn't been merged into master yet). Set up your donation settings using the Donations wizard. Add the Donation block to a page. Use the block to try and purchase a donation. You should get redirected to Checkout page.

**Simple version**: Go to _Products > Add New_ in WP Admin. Create a Simple Product. Add it to your cart and go to the Checkout. 

_If you need help setting WC up to see the Create an Account section or need other help setting things up, ping me on Slack and I'll walk you through it. Some WC steps have not been automated yet._

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by rickybanister** [comment]
@laurelfulford your edits help a bunch. I noticed the mix of serif and sans-serif type on the page from the payment summary grid to the form below. @claudiulodro could we standardize that more? Is it possible to inherit the type from the theme instead of specifying anything except sizing locally here?

**Comment by laurelfulford** [comment]
> I noticed the mix of serif and sans-serif type on the page from the payment summary grid to the form below.

I think this is coming from a weird decision I made in the theme, switching the table to use a sans-serif font when the rest of the body uses a serif font. I can undo that in a separate PR -- I thought it would help differentiate the tables a bit more, but I expect it'll also create other weird situations like this instead.

**Comment by laurelfulford** [comment]
I pushed up the styles that update the table in 364f51af6dc50c35dbca5aa3d9e5fcef39f7bb08, so the only remaining things I can see for this PR would be the minor code tweaks noted above. 

I'll follow up once I have a PR that resets the tables in the theme to use the same font as the rest of the body text.

File: inc/woocommerce.php [hunk 1]
```diff
@@ -80,6 +80,53 @@ function newspack_woo_custom_colors_css( $css, $primary_color ) {
...
+ */
+remove_action( 'woocommerce_checkout_order_review', 'woocommerce_checkout_payment', 20 );
+add_action( 'woocommerce_checkout_after_customer_details', 'woocommerce_checkout_payment' );
+
+/**
+ * Add heading above payment info form.
+ */
+function newspack_woo_payment_heading() {
+	?>
+	<h3><?php echo esc_html__( 'Payment info', 'newspack' ); ?></h3>
```
[the remark this answers is not in the document]
Comment (claudiulodro) [reply inc/woocommerce.php]: https://github.com/Automattic/newspack-theme/pull/153/commits/668a23abaaad6912f606e30ef9f86b06584a867a

File: woocommerce/checkout/form-checkout.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+do_action( 'woocommerce_before_checkout_form', $checkout );
+
+// If checkout registration is disabled and not logged in, the user cannot checkout.
+if ( ! $checkout->is_registration_enabled() && $checkout->is_registration_required() && ! is_user_logged_in() ) {
+	echo esc_html( apply_filters( 'woocommerce_checkout_must_be_logged_in_message', __( 'You must be logged in to checkout.', 'woocommerce' ) ) );
```
[the remark this answers is not in the document]
Comment (claudiulodro) [reply woocommerce/checkout/form-checkout.php]: https://github.com/Automattic/newspack-theme/pull/153/commits/668a23abaaad6912f606e30ef9f86b06584a867a

File: inc/woocommerce.php [hunk 3]
```diff
@@ -80,6 +80,53 @@ function newspack_woo_custom_colors_css( $css, $primary_color ) {
...
+
+/**
+ * Add heading above checkout account creation form.
+ */
+function newspack_woo_account_registration_heading() {
+	$checkout = WC_Checkout::instance();
+
+	if ( $checkout->get_checkout_fields( 'account' ) ) :
+		?>
+		<h3><?php echo esc_html__( 'Create an account', 'newspack' ); ?></h3>
```
[the remark this answers is not in the document]
Comment (claudiulodro) [reply inc/woocommerce.php]: https://github.com/Automattic/newspack-theme/pull/153/commits/668a23abaaad6912f606e30ef9f86b06584a867a

**Comment by claudiulodro** [comment]
Thanks for the review and the styling. I agree; your changes are a big improvement. 👍 

I've made your requested revisions, so I think this should be good to merge.

**Comment by laurelfulford** [comment]
LGTM! 🚢

**Pull request merged**